server/src/core/search/search_executor/preprocess.rs

```diff
@@ -0,0 +1,297 @@
+
+        // There is a special case when up to 3 alphabetic chars are followed by a numeric part.
+        // This is intended to split up strings like "MW1250".
+        if c.is_numeric() && 0 < alphabetic_counter && alphabetic_counter <= 3 {
+            tokens.push(InputToken {
+                s: q.get(token_start..=(i - 1))
+                    .unwrap()
+                    .trim_end()
+                    .to_lowercase(),
+                regular_split: false,
+                closed: true,
+            });
+
+            token_start = i;
+        }
+
+        if (!within_quotes && c.is_whitespace() && i > token_start) ||  // whitespace
+           ((within_quotes || !c.is_whitespace()) && i+c.len_utf8() == q.len()) ||  // end of string
+           (c == '"')
+        {
+            // end of quotes
+            let raw_token = q.get(token_start..i + c.len_utf8());
+            if let Some(token) = raw_token {
+                tokens.push(InputToken {
+                    // Note: The trim_end also trims within unclosed quotes at the end of the query,
+                    //       but currently I don't think this is an issue.
+                    s: token.trim_end().to_lowercase(),
+                    regular_split: true,
+                    // `closed` indicates whether the token has been closed (by whitespace or quote)
+                    // at the end, when this is the last token. This is relevant because MeiliSearch
+                    // treats whitespace at the end differently, and we might want to imitate that
+                    // behaviour.
+                    closed: !(i + c.len_utf8() == q.len()
+                        && (within_quotes || (c != '"' && !c.is_whitespace()))),
+                });
+            }
+
+            token_start = i + 1;
+        } else if !within_quotes && c.is_whitespace() {
+            //    ^
+            // To avoid empty tokens when there are multiple whitespaces, we need
+            // to move the token start even if there was no split.
+            token_start = i + 1;
+        }
+
+        if c.is_alphabetic() {
+            alphabetic_counter += 1;
+        } else {
+            alphabetic_counter = 0;
+        }
+    }
+
+    tokens
+}
+
+#[cfg(test)]
+mod tokenizer_tests {
+    use super::*;
+
+    fn reg(s: &str) -> InputToken {
+        InputToken {
+            s: s.to_string(),
+            regular_split: true,
+            closed: true,
+        }
+    }
+
+    fn irreg(s: &str) -> InputToken {
+        InputToken {
+            s: s.to_string(),
+            regular_split: false,
+            closed: true,
+        }
+    }
+
+    fn assert_token(q: String, expected: Vec<InputToken>) {
+        assert_eq!(
+            tokenize_input_query(q.clone()),
+            expected,
+            "tokenization for '{}' failed",
+            q
+        );
+    }
+
+    fn assert_tokens(q: &str, mut expected: Vec<InputToken>) {
+        let sqs = format!(" {} ", q);
```

Comment:
is this correct behaviour? For trailing spaces closed is true, while if they are not there it is false..

Reply:
See above, that's how I tried to forward that information to MeiliSearch